src/main/kotlin/lotto/controller/LottoController.kt

```diff
@@ -0,0 +1,89 @@
+package lotto.controller
+
+import lotto.model.*
+import lotto.util.RankType
+import lotto.view.InputView
+import lotto.view.OutputView
+
+class LottoController {
+    fun run() {
+        val money = getPurchaseAmount()
+        val lottoMachine = generateLottoTickets(money)
+        showLottoTickets(lottoMachine)
+        val lotto = inputWinningLottoNumbers()
+        val bonusNumber = inputBonusNumber(lotto)
+        val rank = calculateRank(lotto, bonusNumber, lottoMachine)
+        showResult(money, rank)
+    }
+
+    private fun getPurchaseAmount(): Money {
+        while (true) {
+            try {
+                val inputAmount = tryGettingPurchaseAmount()
+                return Money(inputAmount)
+            } catch (e: IllegalArgumentException) {
+                println(e.message)
+            }
+        }
+    }
+
+    private fun tryGettingPurchaseAmount(): Int {
+        OutputView.printAmountMessage()
+        return InputView.getSingleDigit()
+    }
+
+    private fun generateLottoTickets(money: Money): LottoMachine {
+        OutputView.printPurchaseCountMessage(money.getLottoCount())
+        return LottoMachine(money)
+    }
+
+    private fun showLottoTickets(lottoMachine: LottoMachine) {
+        val lottoTicketDTOList = lottoMachine.generateLottoTicketDTOList()
+        lottoTicketDTOList.forEach { lottoTicketDTO ->
+            OutputView.printTicketMessage(lottoTicketDTO.numbers)
+        }
+    }
+
+    private fun inputWinningLottoNumbers(): Lotto {
+        while (true) {
+            try {
+                val lottoNumbers = tryGettingWinningLottoNumbers()
+                return Lotto(lottoNumbers.map { LottoNumber(it) })
+            } catch (e: IllegalArgumentException) {
+                println(e.message)
+            }
+        }
+    }
+
+    private fun tryGettingWinningLottoNumbers(): List<Int> {
+        OutputView.printWinningNumberMessage()
+        return InputView.getMultipleDigit()
+    }
+
+    private fun inputBonusNumber(lotto: Lotto): BonusNumber {
+        while (true) {
+            try {
+                val number = tryGettingBonusNumber()
+                return BonusNumber(number, lotto)
+            } catch (e: IllegalArgumentException) {
+                println(e.message)
+            }
+        }
+    }
+
+    private fun tryGettingBonusNumber(): Int {
+        OutputView.printBonusNumberMessage()
+        return InputView.getSingleDigit()
+    }
+
+    private fun calculateRank(lotto: Lotto, bonusNumber: BonusNumber, lottoMachine: LottoMachine): Map<RankType, Int> {
+        val rankCalculator = RankCalculator(lotto, bonusNumber, lottoMachine)
+        return rankCalculator.calculateRank()
+    }
```

Comment:
사소하지만 이 두 방식중 어떤 방식이 더 가독성이 좋아보이는지 인협님 의견이 궁금합니다!!🤔
전 2개까지는 괜찮은거 같은데 3개부터는 줄바꿈해주는 방식이 더 좋았던거 같아요!!
```suggestion
    private fun calculateRank(
        lotto: Lotto,
        bonusNumber: BonusNumber,
        lottoMachine: LottoMachine
    ): Map<RankType, Int> {
        val rankCalculator = RankCalculator(lotto, bonusNumber, lottoMachine)
        return rankCalculator.calculateRank()
    }
```

Reply:
이런 부분까지는 신경 쓰지 못했던 것 같네요
꼼꼼한 지적 감사합니다 ㅎㅎ